server/src/main/java/org/elasticsearch/indices/SystemIndexDescriptor.java

```diff
@@ -289,13 +334,47 @@ public String checkMinimumNodeVersion(String cause, Version actualMinimumNodeVer
         return null;
     }
 
-    // TODO: getThreadpool()
-    // TODO: Upgrade handling (reindex script?)
-
     public static Builder builder() {
         return new Builder();
     }
 
+    /**
+     * The specific type of system index that this descriptor represents. System indices have three defined types, which is used to
+     * control behavior. Elasticsearch itself has system indices that are necessary for features of Elasticsearch; these system indices
+     * are referred to as internal system indices. Internal system indices are always managed indices that Elasticsearch itself manages.
```

Comment:
I think it is worth mentioning that plugins, even third-party plugins, can implement internal indices. A plugin developer trying to implement SystemIndicesPlugin might look here to decide what type of index to choose.